### PR TITLE
Make golint and govet work in CI

### PR DESCRIPTION
**Pull request by rji**
golint and govet were working for local development, but weren't working reliably in CI. Hopefully this fixes it.

### CODE REVIEW

**Comment by rji** [comment]
This also resolves a number of issues found by golint and govet in order to get a green build.

**Comment by malnick** [comment]
:+1:

File: collector/mesos_agent/mesos_agent.go [hunk 1]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package mesos_agent
```
Comment (malnick) [inline collector/mesos_agent/mesos_agent.go]: Why are we renaming this package? It's common convention in golang to underscore, not camelcase, the package name: https://blog.golang.org/package-names

File: config.go [hunk 2]
```diff
@@ -50,9 +50,9 @@ type Config struct {
 // portion of this project. That is, the code responsible for querying Mesos,
 // et. al to gather metrics and send them to a "producer".
 type CollectorConfig struct {
-	HTTPProfiler bool                             `yaml:"http_profiler"`
-	Node         *node.NodeCollector              `yaml:"node,omitempty"`
-	MesosAgent   *mesos_agent.MesosAgentCollector `yaml:"mesos_agent,omitempty"`
```
Comment (malnick) [inline config.go]: See underscore vs. package name comment above. For variables, camel case is fine. However, for files and package names the convention is lower case with underscores.

File: http_helpers/http_helpers.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package http_helpers
```
Comment (malnick) [inline http_helpers/http_helpers.go]: Naming convention again, this should be underscores and lowercase

File: schema/generator.go [hunk 4]
```diff
@@ -132,7 +132,7 @@ func main() {
 	if err != nil {
 		log.Fatalf("Couldn't open output %s: %s", *outfileFlag, err)
 	}
-	if _, err = fmt.Fprint(outfile, `package metrics_schema
```
Comment (malnick) [inline schema/generator.go]: Same here.

**Comment by malnick** [comment]
Please see https://blog.golang.org/package-names for package naming conventions. Another good resource is net/http, which has a similar sub package structure as this project. The naming should be lowercase and underscores instead of camel case.

**Comment by rji** [comment]
@malnick -- the package names have been changed based on recommendations from [golang/golint](https://github.com/golang/lint). The blog post you linked to is over a year old. According to the latest documentation at https://golang.org/doc/effective_go.html#package-names:

> By convention, packages are given lower case, single-word names; there should be no need for underscores or mixedCaps. [...] Another convention is that the package name is the base name of its source directory; the package in src/encoding/base64 is imported as "encoding/base64" but has name base64, not encoding_base64 and not encodingBase64.

By that convention, both of us are incorrect. But I believe the [MixedCase section of the Effective Go docs](https://golang.org/doc/effective_go.html#mixed-caps) are the tie breaker here:

> Finally, the convention in Go is to use MixedCaps or mixedCaps rather than underscores to write multiword names.

Golint complains about underscores in package names, and not mixedCase. For consistency, this project has implemented all recommendations from golint up to this point; I'd like to continue that.

**Comment by rji** [comment]
Specifically: https://github.com/golang/lint/blob/206c0f020eba0f7fbcfbc467a5eb808037df2ed6/lint.go#L538-L540 (golang/lint at 206c0f0, which is the SHA for HEAD as of this comment)

**Comment by rji** [comment]
Alternately, I'm happy to push up a commit that downcases all package names if you don't like mixedCase. That should make the linter pass as well.

**Comment by malnick** [comment]
The current post you have won me over, mixed caps it is.

**Comment by rji** [comment]
Cool, let me change up the filenames to match the package names for consistency. Then this will be ready for review/merge.

**Comment by rji** [comment]
@malnick -- I think you might like this one :) none of the packages are mixedCase nor do they contain underscores now :smile:

**Comment by rji** [comment]
Ah, looks like it needs a rebase though.

**Comment by malnick** [comment]
Love it, :ship: it.

**Pull request merged**